src/main/java/org/junit/runner/OrderWith.java

```diff
@@ -0,0 +1,26 @@
+package org.junit.runner;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.junit.runner.manipulation.Ordering;
+
+/**
+ * When a test class is annotated with <code>&#064;OrderWith</code> or extends a class annotated
+ * with <code>&#064;OrderWith</code>, JUnit will order the tests in the test class (and child
+ * test classes, if any) using the ordering defined by the {@link Ordering} class.
+ *
+ * @since 4.13
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+@Inherited
+public @interface OrderWith {
```

Comment:
@dsaff I don't see how what you are proposing would be 100% impossible. You could write your own ordering that reads the file the first time `apply()` is called. If the location of the file should be specified by the test writer, then your `Ordering` could use a custom annotation.

I'd hate to add more broiler-plate because of a possibly rare use case. Although I've personally never been bothered by adding a method with an annotation that returns something, other people have objected loudly to this (for example, with Rules).

The problem with passing a `Description` when you create an ordering is that an ordering applies to all children. Luckily, you can get a `Description` from a `Runner` so a custom `Ordering` do a different algorithm based on the `Description` inside of `Ordering.apply()`. I could be convinced that `Ordering.order()` should also take in the parent `Description`.